src/adapter/src/rbac.rs

```diff
@@ -442,35 +459,34 @@ fn ownership_err(
     }
 }
 
+/// Generates the privileges required to execute a given plan.
+///
 /// The result of this function is a set of tuples of the form
 /// (What object the privilege is on, What privilege is required, Who must possess the privilege).
-///
-/// We use a [`BTreeSet`] because we may generate duplicate privileges while traversing through the
-/// plan and used objects.
 fn generate_required_privileges(
     catalog: &impl SessionCatalog,
     plan: &Plan,
     depends_on: &Vec<GlobalId>,
     role_id: RoleId,
-) -> BTreeSet<(ObjectId, AclMode, RoleId)> {
+) -> Vec<(ObjectId, AclMode, RoleId)> {
     match plan {
         Plan::CreateConnection(plan) => {
-            let mut privileges = BTreeSet::from([(
+            let mut privileges = vec![(
                 plan.name.qualifiers.clone().into(),
                 AclMode::CREATE,
                 role_id,
-            )]);
+            )];
             privileges.extend(generate_item_usage_privileges(catalog, depends_on, role_id));
             privileges
         }
         Plan::CreateSchema(plan) => match plan.database_spec {
-            ResolvedDatabaseSpecifier::Ambient => BTreeSet::new(),
+            ResolvedDatabaseSpecifier::Ambient => Vec::new(),
```

Comment:
What is an Ambient and are we using it anywhere?

Reply:
The catalog schemas (`pg_catalog`, `mz_catalog`, etc.) do not belong in any one database. Instead they implicitly belong in all databases. We use "Ambient" to describe the database of these schema.